src/components/templates/Auditorium/components/SectionPreviews/SectionPreviews.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useMemo } from "react";
+import { useHistory } from "react-router";
+import classNames from "classnames";
+
+import { AuditoriumSizes } from "types/auditorium";
+import { AuditoriumVenue } from "types/venues";
+
+import { useAuditoriumSections } from "hooks/auditoriumSections";
+import { useVenueId } from "hooks/useVenueId";
+
+import { SectionPreview } from "../SectionPreview";
+import { Video } from "../Video";
+
+import "./SectionPreviews.scss";
+
+export interface SectionPreviewsProps {
+  venue: AuditoriumVenue;
+}
+
+export const SectionPreviews: React.FC<SectionPreviewsProps> = ({ venue }) => {
+  const { iframeUrl } = venue;
+
+  const venueId = useVenueId();
+  const history = useHistory();
+  const { auditoriumSections } = useAuditoriumSections(venueId);
+
+  const sectionsAmount = auditoriumSections.length;
+
+  const auditoriumSize: AuditoriumSizes = useMemo(() => {
+    if (sectionsAmount <= 4) return AuditoriumSizes.SMALL;
+
+    if (sectionsAmount > 4 && sectionsAmount <= 10) {
+      return AuditoriumSizes.MEDIUM;
+    }
+
+    return AuditoriumSizes.LARGE;
+  }, [sectionsAmount]);
+
+  const sectionPreviews = auditoriumSections.map((section) => (
+    <SectionPreview
+      key={section.id}
+      section={section}
+      onClick={() =>
+        history.push(`${history.location.pathname}/section/${section.id}`)
```

Comment:
Can we just use a relative pathname here rather than needing to prefix with `history.location.pathname`?

---

Also, it might be worth extracting just the bits we need as `const`s rather than accessing them directly from the `history` object so that we can minimize the things that will break memo on this (once its memo'd, as suggested below)

Reply:
Instead of doing this with `onClick` would `Link` or similar make sense to use here at all?

- https://reactrouter.com/web/api/Link

Reply:
As a rule of thumb we should definitely use Link, but in this case specifically, I would prefer to use history(maybe in a more secure way(using a utility func to create a link?), because we have a few usecases for that).

I've changed the logic + added a small note

Reply:
changed here https://github.com/sparkletown/sparkle/pull/1251/commits/adf6f1b18663d0e43209d78822eb78a1f85af80d